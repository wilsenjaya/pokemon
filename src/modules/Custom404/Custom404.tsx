import type { NextPage } from 'next';
import Image from 'next/image';
import { useRouter, NextRouter } from 'next/router';
import { BsArrowLeft } from 'react-icons/bs';
import { Flex, Box, Text, Heading, Button } from '@chakra-ui/react';

import pokeBall from '../../../public/assets/pokeball.png';

export const handleClick = (router: NextRouter) => () => router.push('/');

const renderBig4Text = () => (
  <Heading fontSize={['6rem', '11rem']} color="#cdcdcd" marginX="1rem">
    4
  </Heading>
);

const Custom404: NextPage = () => {
  const router = useRouter();

  return (
    <Flex justify="center" alignItems="center" sx={{ minHeight: '100vh' }}>
      <Box
        maxW="xl"
        borderWidth="1px"
        borderRadius="lg"
        overflow="hidden"
        margin={4}
      >
        <Box p={5}>
          <Box display="flex" alignItems="center">
            {renderBig4Text()}
            <Image src={pokeBall} alt="0" width={130} height={130} />
            {renderBig4Text()}
          </Box>
          <Box
            display="flex"
            flexDir="column"
            alignItems="center"
            justifyContent="center"
          >
            <Heading fontSize="2xl" color="black" marginTop={2}>
              Page Not Found
            </Heading>
            <Text fontSize="l" color="grey" marginTop={2}>
              You look lost on your journey!
            </Text>
            <Button
              colorScheme="red"
              leftIcon={<BsArrowLeft />}
              marginTop={3}
              onClick={handleClick(router)}
            >
              Go Back Home
            </Button>
          </Box>
        </Box>
      </Box>
    </Flex>
  );
};

export default Custom404;
