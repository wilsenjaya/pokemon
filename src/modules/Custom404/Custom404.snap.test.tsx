import renderer from 'react-test-renderer';

import Custom404 from './Custom404';

describe('#Custom404 Page', () => {
  it('should renders correctly', () => {
    const tree = renderer.create(<Custom404 />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
