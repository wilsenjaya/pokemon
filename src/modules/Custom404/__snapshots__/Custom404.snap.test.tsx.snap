// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Custom404 Page should renders correctly 1`] = `
<div
  className="css-1o0ed15"
>
  <div
    className="css-1336sx9"
  >
    <div
      className="css-1g360fb"
    >
      <div
        className="css-70qvj9"
      >
        <h2
          className="chakra-heading css-yn02dh"
        >
          4
        </h2>
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "width": "initial",
            }
          }
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27130%27%20height=%27130%27/%3e"
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            />
          </span>
          <img
            alt="0"
            data-nimg="intrinsic"
            decoding="async"
            onError={[Function]}
            onLoad={[Function]}
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="0"
              data-nimg="intrinsic"
              decoding="async"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=384&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=384&q=75 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </span>
        <h2
          className="chakra-heading css-yn02dh"
        >
          4
        </h2>
      </div>
      <div
        className="css-u6v8er"
      >
        <h2
          className="chakra-heading css-2ryi4e"
        >
          Page Not Found
        </h2>
        <p
          className="chakra-text css-159gddx"
        >
          You look lost on your journey!
        </p>
        <button
          className="chakra-button css-1h8u2ox"
          onClick={[Function]}
          type="button"
        >
          <span
            className="chakra-button__icon css-1wh2kri"
          >
            <svg
              aria-hidden={true}
              fill="currentColor"
              focusable={false}
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 8a.5.5 0 0 0-.5-.5H2.707l3.147-3.146a.5.5 0 1 0-.708-.708l-4 4a.5.5 0 0 0 0 .708l4 4a.5.5 0 0 0 .708-.708L2.707 8.5H14.5A.5.5 0 0 0 15 8z"
                fillRule="evenodd"
              />
            </svg>
          </span>
          Go Back Home
        </button>
      </div>
    </div>
  </div>
</div>
`;
